vllm/v1/engine/output_processor.py

```diff
@@ -133,7 +134,8 @@ def process_outputs(
 
         request_outputs: List[RequestOutput] = []
         reqs_to_abort: List[str] = []
-        iteration_stats = IterationStats(self.log_stats)
+        if not iteration_stats:
+            iteration_stats = IterationStats(self.log_stats)
```

Comment:
Why pass in an iteration stats just to override it? Is the arg more like a bool?

Reply:
It's not overridden - it gets created here only if it isn't passed in (or is `None`).

This was just to keep the change minimal since the method is also used elsewhere. Now that it's called multiple times per iteration, we need to update the stats rather than only returning them.

Reply:
Sorry I misunderstood the negation 😓 thanks